starter_code/src/movies.js

```diff
@@ -1,13 +1,44 @@
+"use strict";
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
 
 
 // Get the average of all rates with 2 decimals 
 
+//var decimalNumber = parseFloat(movies[rate].toFixed(2));
+
+function ratesAverage(array){
+    var total = array.reduce(function (accumulator, item) {
+        return accumulator + Number(item.rate); 
+    },0).toFixed(2);
+    return total/array.length;
+}
+ratesAverage(movies);
 
 // Get the average of Drama Movies
 
 
+var total = array.reduce(function(accumulator, item) {
+    if (item.genre === "Drama") {
+        return 1;
+    } else {
+        return -1;
+    }
+});
+
+
+function dramaMoviesRate(array) {
+    var sum = array.reduce(function (accumulator, item) {
```

Comment:
Don't you feel like you are repeating yourselves? After filtering the array (see previous comment), you can reuse the function that you write before (ratesAverage) to calculate the average.